plugins/user/cronjobs

```diff
@@ -0,0 +1,193 @@
+#!/usr/bin/env python3
+# -*- python -*-
+"""
+=head1 NAME
+
+Plugin to monitor the number of cronjobs running per user, gathering data from
+syslog.
+
+=head1 INSTALLATION
+
+Usage: Place in /etc/munin/plugins/ (or link it there using ln -s)
+
+=head1 CONFIGURATION
+
+Add this to your /etc/munin/plugin-conf.d/munin-node:
+
+=over 2
+
+      [cronjobs]
+      user root
+      env.syslog_path /var/log/syslog # default value
+      env.cron_ident_regex crond? # for finding cron entries in the syslog,
+                                    case-insensitive (default value)
+
+=back
+
+The plugin needs to run as root in order to read from syslog.
+
+=head1 HISTORY
+
+2019-09-09: v 1.0 pcy <pcy.ulyssis.org>: created
+
+=head1 USAGE
+
+Parameters understood:
+
+  config   (required)
+  autoconf (optional - used by munin-config)
+
+=head1 MAGIC MARKERS
+
+#%# family=auto
+#%# capabilities=autoconf
+"""
+
+
+from datetime import datetime, timedelta, timezone
+import os
+import re
+import shutil
+import socket
+import sys
+import time
+
+
+syslog_path = os.getenv('syslog_path', '/var/log/syslog')
+cron_ident_regex = os.getenv('cron_ident_regex', 'crond?')
+
+# expected format:
+# <date> <hostname> CRON[<pid>]: (<user>) CMD ...
+# example:
+# Aug 16 22:00:01 zap CRON[23060]: (root) CMD (/usr/local/bin/dnsconfig -s)
+cron_syslog_regex = re.compile('^.* %s %s\[[0-9]*\]: \((.*?)\)'
+                               % (socket.gethostname(), cron_ident_regex),
+                               re.I)
+
+STATEFILE = os.getenv('MUNIN_STATEFILE')
+STATE = None
+
+
+if os.path.isfile(STATEFILE):
+    with open(STATEFILE, 'r') as f:
+        if sys.version_info >= (3, 7):
```

Comment:
I guess an epoch value might work.

Also, I just thought, the log file position: wouldn't I also have to check and compare the mtime against the last run, because sometimes after a logrotate, the file might be large enough again for the seek position to be ok. (Won't happen very often, but still, not completely impossible.) So I'd have to store a timestamp regardless.